mockgen/parse.go

```diff
@@ -641,13 +641,16 @@ func parsePackageImport(srcDir string) (string, error) {
 		}
 	}
 	// fall back to GOPATH mode
-	goPath := os.Getenv("GOPATH")
-	if goPath == "" {
+	goPaths := os.Getenv("GOPATH")
+	if goPaths == "" {
 		return "", fmt.Errorf("GOPATH is not set")
 	}
-	sourceRoot := filepath.Join(goPath, "src") + string(os.PathSeparator)
-	if !strings.HasPrefix(srcDir, sourceRoot) {
-		return "", errOutsideGoPath
+	goPathList := strings.Split(goPaths, ":")
```

Comment:
This assumes UNIX.

> The GOPATH environment variable lists places to look for Go code.
On Unix, the value is a colon-separated string.
On Windows, the value is a semicolon-separated string.
On Plan 9, the value is a list.